### PR TITLE
Adds ships and helm console to the game

### PR DESCRIPTION
**Pull request by SuperNovaa41**
THIS IS NOT DONE.

Adds the helm console, with a very makeshift UI for now. As all it has is the ship viewer and the movement controls.
Aswell, it just adds a basic proc for creating ships on the overmap, and automatically linking them to their map.

### CODE REVIEW

File: voidcrew/edits/mobs/new_player.dm [hunk 1]
```diff
@@ -24,7 +24,7 @@
 			return LateChoices()
 
 		to_chat(usr, span_danger("Your [initial(template.name)] is being prepared. Please be patient!"))
-		var/obj/docking_port/mobile/target = SSshuttle.load_template(new template) //voidcrew todo: make ships spawn properly
+		var/obj/docking_port/mobile/voidcrew/target = SSshuttle.load_template(new template) //voidcrew todo: make ships spawn properly
```
Comment (JohnFulpWillard) [inline voidcrew/edits/mobs/new_player.dm:27]:
```suggestion
		var/obj/docking_port/mobile/voidcrew/target = SSshuttle.load_template(new template)
```

File: voidcrew/modules/overmap/code/controllers/subsystem/overmap.dm [hunk 2]
```diff
@@ -67,7 +67,18 @@ SUBSYSTEM_DEF(overmap)
...
+/datum/controller/subsystem/overmap/proc/get_unused_overmap_square(thing_not_to_have = /obj/structure/overmap, tries = MAX_OVERMAP_PLACEMENT_ATTEMPTS, force = FALSE)
+	var/turf/turf_to_return
+	for (var/_ in 1 to tries)
+		turf_to_return = pick(block(locate(OVERMAP_LEFT_SIDE_COORD, OVERMAP_SOUTH_SIDE_COORD, OVERMAP_Z_LEVEL), locate(OVERMAP_RIGHT_SIDE_COORD, OVERMAP_NORTH_SIDE_COORD, OVERMAP_Z_LEVEL))) // todo : see if this is expensive
+		if (locate(thing_not_to_have) in turf_to_return)
+			continue
+		return turf_to_return
+	if (!force)
+		turf_to_return = null
+	return turf_to_return
```
Comment (JohnFulpWillard) [inline voidcrew/modules/overmap/code/controllers/subsystem/overmap.dm:79]: What does this do?

File: voidcrew/modules/overmap/code/modules/overmap/_overmap.dm [hunk 3]
```diff
@@ -3,5 +3,6 @@
 	desc = "An unknown celestial object."
 	icon = 'voidcrew/modules/overmap/icons/effects/overmap.dmi'
 	icon_state = "object"
+	var/integrity
```
Comment (JohnFulpWillard) [inline voidcrew/modules/overmap/code/modules/overmap/_overmap.dm:6]: doc?

File: voidcrew/modules/overmap/code/modules/overmap/ship.dm [hunk 4]
```diff
@@ -46,13 +47,112 @@
...
 
+	/**
+	 * Movement stuff
+	 */
+	var/y_thrust = 0
+	var/x_thrust = 0
+	// Stuff needed to render the map
+	var/map_name
+	var/atom/movable/screen/map_view/cam_screen
+	var/list/cam_plane_masters
```
Comment (JohnFulpWillard) [inline voidcrew/modules/overmap/code/modules/overmap/ship.dm:58]:
```suggestion
	var/list/cam_plane_masters = list()
```

File: voidcrew/modules/overmap/code/modules/overmap/ship.dm [hunk 5]
```diff
@@ -46,13 +47,112 @@
...
 
 	display_name = "[faction_prefix] [name]"
 
+	map_name = "overmap_[REF(src)]_map"
+	cam_screen = new
+	cam_screen.name = "screen"
+	cam_screen.assigned_map = map_name
+	cam_screen.del_on_map_removal = FALSE
+	cam_screen.screen_loc = "[map_name]:1,1"
+	cam_plane_masters = list()
```
Comment (JohnFulpWillard) [inline voidcrew/modules/overmap/code/modules/overmap/ship.dm:74]:
```suggestion
```

File: voidcrew/modules/overmap/code/modules/overmap/ship.dm (hunk 4 above)
Comment (JohnFulpWillard) [inline voidcrew/modules/overmap/code/modules/overmap/ship.dm:58]: docs?

File: voidcrew/modules/overmap/code/modules/overmap/_overmap.dm (hunk 3 above)
Comment (SuperNovaa41) [reply voidcrew/modules/overmap/code/modules/overmap/_overmap.dm:6]: not needed (for now), removing

**Pull request merged**